pkg/cmd/variable/shared/shared.go

```diff
@@ -0,0 +1,317 @@
+package shared
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"sort"
+	"strings"
+	"time"
+
+	"github.com/cli/cli/v2/api"
+	"github.com/cli/cli/v2/internal/config"
+	"github.com/cli/cli/v2/internal/ghrepo"
+	"github.com/cli/cli/v2/internal/text"
+	"github.com/cli/cli/v2/pkg/iostreams"
+)
+
+type Visibility string
+
+const (
+	All      = "all"
+	Private  = "private"
+	Selected = "selected"
+)
+
+type App string
+
+const (
+	Actions = "actions"
+	Unknown = "unknown"
+)
+
+func (app App) String() string {
+	return string(app)
+}
+
+func (app App) Title() string {
+	return text.Title(app.String())
+}
+
+type VariableEntity string
+
+const (
+	Repository   = "repository"
+	Organization = "organization"
+	Environment  = "environment"
+)
+
+type VariablePayload struct {
+	Name         string  `json:"name,omitempty"`
+	Value        string  `json:"value,omitempty"`
+	Visibility   string  `json:"visibility,omitempty"`
+	Repositories []int64 `json:"selected_repository_ids,omitempty"`
+}
+
+type repoNamesResult struct {
+	Ids []int64
+	Err error
+}
+
+type PostPatchOptions struct {
```

Comment:
```suggestion
type SetOptions struct {
```

We like to keep the definition of the options structs in the same file as the command typically unless there is a good reason not to.

Reply:
done.